### PR TITLE
Praj/1901

### PR DESCRIPTION
**Pull request by PrajaktaPurohit**
### Description

This will log more information in case of a failure in reindex due to slow es.
Adding a try/catch around the case in add_batch error out immediately on the try.

```
root@ip-10-0-16-115:~# chef-server-ctl reindex org4elastic -w -t -d
The search queue mode is currently configured to be "batch."
Ignoring "wait" option.
Reindexing orgs: ["org4elastic"]
- Disabling the Chef API.
- Enqueueing data for indexing.
Removing all index entries for organization 'org4elastic'...
Sending all data for organization 'org4elastic' to be indexed again.  It may take some time before everything is available via search.
escript: exception error: no case clause matching
                 {badrpc,
                     {'EXIT',
                         {[{{node,<<"8cbecf9d307ab1350f471e419021fdf3">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,
                                ["error",44,
                                 [123,
                                  ["conn_failed",44,
                                   [123,["error",44,"econnrefused"],125]],
                                  125]],
                                125]],
                              125]]},
                           {{node,<<"8cbecf9d307a15bd7251427023d36232">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,
                                ["error",44,
                                 [123,
                                  ["conn_failed",44,
                                   [123,["error",44,"econnrefused"],125]],
                                  125]],
                                125]],
                              125]]},
                           {{node,<<"8cbecf9d307a919e0513878158c70867">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]},
                           {{node,<<"8cbecf9d307a9e49547d148e85d20801">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]},
                           {{node,<<"8cbecf9d307a60b446a77e1f3ce183b2">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]},
                           {{node,<<"8cbecf9d307aa4c52d0b933284c23b03">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,
                                ["error",44,
                                 [123,
                                  ["conn_failed",44,
                                   [123,["error",44,"econnrefused"],125]],
                                  125]],
                                125]],
                              125]]},
                           {{node,<<"8cbecf9d307adce89b870956e07c5431">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]},
                           {{node,<<"8cbecf9d307aac3c161fa156d7db30a5">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]},
                           {{node,<<"8cbecf9d307a91a3ccfcb0bc22b8941f">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,
                                ["error",44,
                                 [123,
                                  ["conn_failed",44,
                                   [123,["error",44,"econnrefused"],125]],
                                  125]],
                                125]],
                              125]]},
                           {{node,<<"8cbecf9d307a04230b98d8a1b8a438c8">>,
                                <<"chef_a695f596e0a777b5c90d8cbecf9d307a">>},
                            [101,114,114,111,114,58,32,
                             [123,
                              ["error",44,
                               [123,["error",44,"sel_conn_closed"],125]],
                              125]]}],
                          [{chef_reindex,add_batch,2,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/chef_reindex.erl"},
                                {line,248}]},
                           {chef_reindex,send_to_index_queue,5,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/chef_reindex.erl"},
                                {line,234}]},
                           {chef_reindex,'-batch_reindex/6-fun-0-',6,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/chef_reindex.erl"},
                                {line,192}]},
                           {chefp,batch_fold0,4,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/chef_db/src/chefp.erl"},
                                {line,47}]},
                           {chef_reindex,'-reindex/2-lc$^2/1-0-',3,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/chef_reindex.erl"},
                                {line,51}]},
                           {chef_reindex,reindex,2,
                               [{file,
                                    "/var/cache/omnibus/chef-server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/chef_reindex.erl"},
                                {line,51}]},
                           {rpc,'-handle_call_call/6-fun-0-',5,
                               [{file,"rpc.erl"},{line,197}]}]}}}
Failed to enqueue data for org4elastic!
- Re-enabling the Chef API
5.780380019 seconds to reindex.
```

The error looks better just with the case statement:

```
root@ip-10-0-16-115:~# chef-server-ctl reindex org4elastic -w -t -d
The search queue mode is currently configured to be "batch."
Ignoring "wait" option.
Reindexing orgs: ["org4elastic"]
- Disabling the Chef API.
- Enqueueing data for indexing.
Removing all index entries for organization 'org4elastic'...
Sending all data for organization 'org4elastic' to be indexed again.  It may take some time before everything is available via search.
Reindexing the organization "org4elastic" has failed.
The following objects failed to be reindexed in the last batch:
	node[8cbecf9d307a1ae0e01c1ecfec833c5c]: error: {error,{error,sel_conn_closed}}
	node[8cbecf9d307aea7a9629c52d5632c4cf]: error: {error,{error,sel_conn_closed}}
	node[8cbecf9d307acf8c9458e36b868bfcac]: error: {error,{error,sel_conn_closed}}
	node[8cbecf9d307ae58155dfc22d62d1db3f]: error: {error,{error,sel_conn_closed}}
	node[8cbecf9d307a310902ac8c774aaecd4e]: error: {error,{error,{conn_failed,{error,econnrefused}}}}
	node[8cbecf9d307a2ad679281863474633de]: error: {error,
Failed to enqueue data for org4elastic!
- Re-enabling the Chef API
244.678836035 seconds to reindex.
```

### Issues Resolved

https://github.com/chef/chef-server/issues/1901

Integration build:
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/1067#69b8729a-8a67-4213-961e-caa99f337407

### Check List

- [x] All buildkite tests pass
- [x] Full omnibus build and tests in buildkite pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [x] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl [hunk 1]
```diff
@@ -235,9 +235,10 @@ send_to_index_queue(OrgName, OrgId, Index, SerializedObjects, NameIdDict) ->
     {Res, Missing}.
 
 add_batch(Batch, OrgName) ->
-    case oc_chef_object_db:add_batch_to_solr(Batch) of
-        ok -> ok;
-        {error, Failures} ->
+    try oc_chef_object_db:add_batch_to_solr(Batch) of
+        ok -> ok
+    catch
+        error:Failures ->
```
Comment (marcparadise) [inline src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl:241]: I don't think the error:Failures catch will catch an `{error, Failures}` tuple returned from add_batch_to_solr.

File: src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl [hunk 2]
```diff
@@ -255,8 +258,12 @@ log_failures(OrgName, [Failure|Rest]) ->
 humanize_failures([], Acc) ->
     Acc;
 humanize_failures([H|T], Acc) ->
-    {Id, Reason} = H,
-    humanize_failures(T, [{Id, pretty_reason(Reason)} | Acc]).
+    case H of
+        {Id, Reason} -> humanize_failures(T, [{Id, pretty_reason(Reason)} | Acc]);
+        Error -> humanize_failures(T, [{<<"ERROR">>, pretty_reason(Error)} | Acc])
+    end;
+humanize_failures(Failures, []) ->
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl]: Tracing the specs back, it looks like Failures will always be a list.

File: src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl [hunk 3]
```diff
@@ -252,17 +255,26 @@ log_failures(OrgName, [Failure|Rest]) ->
...
 
+-spec humanize_failures(list(), list()) -> list().
 humanize_failures([], Acc) ->
     Acc;
 humanize_failures([H|T], Acc) ->
-    {Id, Reason} = H,
-    humanize_failures(T, [{Id, pretty_reason(Reason)} | Acc]).
+    case H of
+        {Id, Reason} -> humanize_failures(T, [{Id, pretty_reason(Reason)} | Acc]);
+        Error -> humanize_failures(T, [{<<"ERROR">>, pretty_reason(Error)} | Acc])
```
Comment (stevendanna) [inline src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl:264]: Since `<<"ERROR">>` is taking the place of Id, maybe `<<"unknown">>` might be better?

Comment (PrajaktaPurohit) [reply src/oc_erchef/apps/oc_chef_wm/src/chef_reindex.erl]: done :)

**Pull request merged**